salt/transport/rabbitmq.py

```diff
@@ -0,0 +1,1243 @@
+"""
+Rabbitmq transport classes
+This is a copy/modify of zeromq implementation with zeromq-specific bits removed.
+TODO: refactor transport implementations so that tcp, zeromq, rabbitmq share common code
+"""
+import copy
+import errno
+import hashlib
+import logging
+import os
+import signal
+import sys
+import threading
+import weakref
+from random import randint
+from uuid import uuid4
+
+import pika
+from pika import SelectConnection, BasicProperties
+from pika.exchange_type import ExchangeType
```

Comment:
I'm assuming this is the rabbitmq lib - if we end out making pika an optional dependency this will need a `ImportError` guard.

Reply:
Makes sense in general, but without pika this entire module is useless, so the import guard should probably be around salt.transport.rabbitmq